### PR TITLE
AG-12847 more bundle size changes

### PR DESCRIPTION
**Pull request by peterjrreynolds**

### CODE REVIEW

File: packages/ag-grid-enterprise/src/masterDetail/masterDetailModule.ts [hunk 1]
```diff
@@ -22,7 +23,7 @@ export const MasterDetailCoreModule: _ModuleWithoutApi = {
         },
         { name: 'agDetailCellRenderer', classImp: DetailCellRenderer },
     ],
-    controllers: [
+    dynamicBeans: [
         { name: 'detailCellRenderer', classImp: DetailCellRendererCtrl },
```
Comment (AndrewGlazier) [inline packages/ag-grid-enterprise/src/masterDetail/masterDetailModule.ts]: please rename to detailCellRendererCtrl

File: packages/ag-grid-community/src/filter/floating/floatingFilterMapper.ts [hunk 2]
```diff
@@ -17,22 +17,11 @@ export function getDefaultFloatingFilterType(
...
         const floatingFilterTypeMap: { [p: string]: string } = {
-            set: 'agSetColumnFloatingFilter',
             agSetColumnFilter: 'agSetColumnFloatingFilter',
-
-            multi: 'agMultiColumnFloatingFilter',
             agMultiColumnFilter: 'agMultiColumnFloatingFilter',
-
-            group: 'agGroupColumnFloatingFilter',
             agGroupColumnFilter: 'agGroupColumnFloatingFilter',
-
```
Comment (StephenCooper) [inline packages/ag-grid-community/src/filter/floating/floatingFilterMapper.ts:28]: Is this breaking change that needs noting on a Jira even though these have been deprecated for a while.

**Review by StephenCooper — APPROVED** [review]
Happy with changes

File: packages/ag-grid-community/src/grid.ts [hunk 3]
```diff
@@ -211,21 +199,22 @@ export class GridCoreCreator {
...
-    private registerModuleApiFunctions(
+    private registerModuleFeatures(
         context: Context,
         registeredModules: (_ModuleWithApi<any> | _ModuleWithoutApi)[]
     ): void {
+        const registry = context.getBean('registry');
         const apiFunctionService = context.getBean('apiFunctionService');
+
         registeredModules.forEach((module) => {
+            module.userComponents?.forEach(({ name, classImp, params }) =>
```
Comment (AndrewGlazier) [inline packages/ag-grid-community/src/grid.ts]: Can we combine these register functions into one function for registering a module with the registry

File: packages/ag-grid-community/src/filter/floating/floatingFilterMapper.ts (hunk 2 above)
Comment (AndrewGlazier) [reply packages/ag-grid-community/src/filter/floating/floatingFilterMapper.ts:28]: +1

File: packages/ag-grid-community/src/validation/enterpriseModuleNames.ts [hunk 4]
```diff
@@ -11,7 +11,9 @@ export const ENTERPRISE_MODULE_NAMES: Record<EnterpriseModuleName, boolean> = {
     ClipboardModule: true,
     ColumnChooserModule: true,
```
Comment (AndrewGlazier) [inline packages/ag-grid-community/src/validation/enterpriseModuleNames.ts]: This whole object should be a set, should save some bundle size not having all these "true" being defined too

File: packages/ag-grid-community/src/interfaces/iModule.ts [hunk 5]
```diff
@@ -27,8 +28,9 @@ export interface Module {
      */
     validate?: () => ModuleValidationResult;
     beans?: SingletonBean[];
-    controllers?: ControllerMeta[];
+    dynamicBeans?: DynamicBeanMeta[];
```
Comment (AndrewGlazier) [inline packages/ag-grid-community/src/interfaces/iModule.ts:33]: can you document this type?

beans, dynamicBeans, selectors and userComponents

File: packages/ag-grid-enterprise/src/masterDetail/masterDetailModule.ts (hunk 1 above)
Comment (peterjrreynolds) [reply packages/ag-grid-enterprise/src/masterDetail/masterDetailModule.ts]: Done

File: packages/ag-grid-community/src/filter/floating/floatingFilterMapper.ts (hunk 2 above)
Comment (peterjrreynolds) [reply packages/ag-grid-community/src/filter/floating/floatingFilterMapper.ts:28]: Done

File: packages/ag-grid-community/src/grid.ts (hunk 3 above)
Comment (peterjrreynolds) [reply packages/ag-grid-community/src/grid.ts]: Done

File: packages/ag-grid-community/src/validation/enterpriseModuleNames.ts (hunk 4 above)
Comment (peterjrreynolds) [reply packages/ag-grid-community/src/validation/enterpriseModuleNames.ts]: Done

File: packages/ag-grid-community/src/interfaces/iModule.ts (hunk 5 above)
Comment (peterjrreynolds) [reply packages/ag-grid-community/src/interfaces/iModule.ts:33]: Done

**Pull request merged**